src/test/java/StringManipulationTest.java

```diff
@@ -49,101 +55,137 @@ public void testRemoveNthCharacter1() {
+
+        assertEquals("This", sStings[0]);
+        assertEquals("a", sStings[2]);
     }
+
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This string is a very long string");
+        String [] sStings = manipulatedstring.getSubStrings(2, 4);
+
+        assertEquals("string", sStings[0]);
+        assertEquals("a", sStings[2]);
     }
+
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This sting will return the first and last string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 9);
+
+        assertEquals("This", sStings[0]);
+        assertEquals("string", sStings[8]);
     }
+
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This string is the last test string");
+        String [] sStings = manipulatedstring.getSubStrings(5, 7);
+
+        assertEquals("last", sStings[0]);
+        assertEquals("string", sStings[2]);
     }
 
     @Test
-    public void testRestoreString1()
-    {
+    public void testRestoreString1() {
         manipulatedstring.setString("art");
         int [] array;
-        array=new int[]{1,0,2};
+        array = new int[]{1,0,2};
         String restoreString = manipulatedstring.restoreString(array);
-        assertEquals(restoreString, "rat");
+        assertEquals("rat", restoreString);
     }
 
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString2() {
+        manipulatedstring.setString("UnitTest");
+        int [] array;
+        array = new int[]{4,5,6,7,0,1,2,3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals("TestUnit", restoreString);
     }
 
     @Test
-    public void testRestoreString3()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString3() {
+        manipulatedstring.setString("eat");
+        int [] array;
+        array = new int[]{2,0,1};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals("tea", restoreString);
     }
 
     @Test
-    public void testRestoreString4()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString4() {
+        manipulatedstring.setString("cat");
+        int [] array;
+        array=new int[]{1,0,2};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals("act", restoreString);
     }
 
     @Test
-    public void testRestoreString5()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString5() {
+        manipulatedstring.setString("bat");
+        int [] array;
+        array=new int[]{2,1,0};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals("tab", restoreString);
```

Comment:
Same as the comment on line 139, I would recommend changing this test to check if exceptions are being thrown properly